### PR TITLE
More general way of adding System Identifiers to Caliper things (cc204)

### PR DESCRIPTION
**Pull request by ViktorHaag**
This PR demos a more general way of including [system identifiers for things in Caliper (cc204)](https://github.com/IMSGlobal/caliper-central/issues/204); it tracks the three basic entities also changed in PR #217. Some things to note:

- we need a new Caliper "blank node" entity called `SystemIdentifier` (by "blank node", I mean it wouldn't have an `id` field, like `TextPositionSelector`)
- each Caliper Entity would have an added `otherIdentifiers` optional field, with possible value being a list of SystemIdentifiers
- SystemIdentifier thing would need: a required "identifier" field to hold the system ID value, a required "identifierType" field to hold the enumerated value of the type of identifier, and an optional "source" field to indicate who made the identifier (or who is the scope of authority over it)

For enumerated identifierTypes, we'd treat that as an enumerated Caliper vocab, just like we do with roles, etc. I think we probably need these entries in it to start:
- LisSourcedId
- OneRosterSourcedId
- SisSourcedId
- LtiUserId
- LtiContextId
- EmailAddress
- AccountUserName
- Other

As @ojurosevic explained, a platform could know that an ID is a "SIS identifier", but not know if the upstream SIS used OneRoster or LIS to get the value into the platform; this leads me to think we need to have a generic SisSourcedId value as well as more specific ones for LIS and OR...

### CODE REVIEW

File: v1p2/caliperEntityCourseOffering.json [hunk 1]
```diff
@@ -5,6 +5,14 @@
   "courseNumber": "CPS 435",
   "academicSession": "Fall 2016",
   "name": "CPS 435 Learning Analytics",
+  "otherIdentifiers": [
+    {
+      "type": "SystemIdentifier",
+      "identifier": "example.edu:SI182-F16",
+      "identifierType": "LisSourcedId",
+      "source": null,
```
Comment (ViktorHaag) [inline v1p2/caliperEntityCourseOffering.json]: note this (and other null source fields) are placeholders so you can see what the fields for this SystemIdentifier thing would be... we can strip them out when/if we decide this is what we want.

**Comment by bracken** [comment]
I like it. I can't decide what direction I think is best to go in. I like the simpler solution because simple and covers _most_ use cases. This pull request is still quite simple though and covers the other cases that come up. So I'm probably in favour of this one if we can keep it simple like this and not start adding tons of stuff.

File: v1p2/caliperEntityPerson.json [hunk 2]
```diff
@@ -2,6 +2,25 @@
...
+      "type": "SystemIdentifier",
+      "identifier": "jane@example.edu",
+      "identifierType": "EmailAddress",
+      "source": "https://example.edu"
+    },
+    {
+      "type": "SystemIdentifier",
+      "identifier": "https://example.edu/users/554433",
+      "identifierType": "LtiUserId",
+      "source": "07940580-b309-415e-a37c-914d387c1150"
```
Comment (ViktorHaag) [inline v1p2/caliperEntityPerson.json]: Note that this is the "deployment ID" for the LTI deployment that is responsible for this LTI User ID. _Knowing_ that this is what the GUID is for is probably magic, unless we include in a spec that the "source of LTI Identifiers should be the deployment ID for the deployment that owns that ID".

In some cases, LTI identifiers will actually be scoped by deployment; in some cases, though, their actual scope may be across several deployments.

**Comment by ViktorHaag** [comment]
The more I look at the `source` field, the less I like it and the more I question it's usefulness (mostly because for it to have a useful value, I think producing it would come with all sorts of complicated rules... it's basically a system identifier for a system... 8p -- maybe we should say that this is an optional EdApp reference, and leave it at that?

**Comment by ViktorHaag** [comment]
:sigh: yet another rebasing mess; I'm going to close this PR, and re-open a new one so that diffs will be more clear.

**Pull request closed without merging**